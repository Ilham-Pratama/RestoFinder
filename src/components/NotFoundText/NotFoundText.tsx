import React from 'react';
import { View } from 'react-native';
import sharedStyles from 'shared/styles';
import ThemedText from '../ThemedText';

export const DEFAULT_NOT_FOUND_TEXT = "Oops, couldn't find anything ðŸ˜¢";

interface INotFoundText {
  text?: string;
}

const NotFoundText = ({ text = DEFAULT_NOT_FOUND_TEXT }: INotFoundText) => {
  return (
    <View
      accessibilityRole="text"
      style={[sharedStyles.alignItemsCenter, sharedStyles.marginTop1]}>
      <ThemedText style={[sharedStyles.marginVertical3]} color="gray">
        {text}
      </ThemedText>
    </View>
  );
};

export default NotFoundText;
