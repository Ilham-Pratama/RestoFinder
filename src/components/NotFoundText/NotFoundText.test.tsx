import { render } from '@testing-library/react-native';
import React from 'react';
import NotFoundText, { DEFAULT_NOT_FOUND_TEXT } from './NotFoundText';

describe('NotFoundText', () => {
  it('renders component with default text', () => {
    const component = render(<NotFoundText />);

    expect(component.toJSON()).toMatchSnapshot();
    expect(component.getByText(DEFAULT_NOT_FOUND_TEXT).props).toHaveProperty(
      'color',
      'gray'
    );
  });

  it('renders component with custom text', () => {
    const customText = 'Here is your text';
    const component = render(<NotFoundText text={customText} />);

    expect(component.toJSON()).toMatchSnapshot();
    expect(component.getByText(customText).props).toHaveProperty(
      'color',
      'gray'
    );
  });
});
