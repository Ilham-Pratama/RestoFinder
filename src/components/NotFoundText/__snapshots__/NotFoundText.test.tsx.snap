// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundText renders component with custom text 1`] = `
<View
  accessibilityRole="text"
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "marginTop": 5,
      },
    ]
  }
>
  <Text
    color="gray"
    style={
      Array [
        Array [
          Object {
            "marginVertical": 15,
          },
        ],
        Object {
          "color": "gray",
        },
      ]
    }
  >
    Here is your text
  </Text>
</View>
`;

exports[`NotFoundText renders component with default text 1`] = `
<View
  accessibilityRole="text"
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "marginTop": 5,
      },
    ]
  }
>
  <Text
    color="gray"
    style={
      Array [
        Array [
          Object {
            "marginVertical": 15,
          },
        ],
        Object {
          "color": "gray",
        },
      ]
    }
  >
    Oops, couldn't find anything ðŸ˜¢
  </Text>
</View>
`;
